crates/ra_editor/src/typing.rs

```diff
@@ -140,30 +141,71 @@ fn remove_newline(
     offset: TextUnit,
 ) {
     if node.kind() == WHITESPACE && node_text.bytes().filter(|&b| b == b'\n').count() == 1 {
+        // Special case that turns something like:
+        //
+        // ```
+        // my_function({<|>
+        //    <some-expr>
+        // })
+        // ```
+        //
+        // into `my_function(<some-expr>)`
         if join_single_expr_block(edit, node).is_some() {
             return
         }
-        match (node.prev_sibling(), node.next_sibling()) {
-            (Some(prev), Some(next)) => {
-                let range = TextRange::from_to(prev.range().start(), node.range().end());
-                if is_trailing_comma(prev.kind(), next.kind()) {
-                    edit.delete(range);
-                } else if no_space_required(prev.kind(), next.kind()) {
-                    edit.delete(node.range());
-                } else if prev.kind() == COMMA && next.kind() == R_CURLY {
-                    edit.replace(range, " ".to_string());
+
+        if let (Some(prev), Some(next)) = (node.prev_sibling(), node.next_sibling()) {
+            let range = TextRange::from_to(prev.range().start(), node.range().end());
+            if is_trailing_comma(prev.kind(), next.kind()) {
+                // Removes: trailing comma, newline (incl. surrounding whitespace)
+                edit.delete(range);
+            } else if no_space_required(prev.kind(), next.kind()) {
+                // Removes: newline (incl. surrounding whitespace)
+                edit.delete(node.range());
+            } else if prev.kind() == COMMA && next.kind() == R_CURLY {
+                // Removes: comma, newline (incl. surrounding whitespace)
+                // Adds: a single whitespace
+                edit.replace(range, " ".to_string());
+            } else if prev.kind() == COMMENT && next.kind() == COMMENT {
+                // Removes: newline (incl. surrounding whitespace), start of the next comment
+
+                // FIXME: I guess it is safe to unwrap here? A comment always has text, right?
```

Comment:
Right, though we should really add a typed wrapper for Comments. See this PR for a similar wrapper for Lifetime: https://github.com/rust-analyzer/rust-analyzer/commit/15f15d92eb4d6ab791047eefbd6dd9b2baba1140#diff-e1e05a48de544b7b334e42793e36541dR478